loader/windows/icd_windows_apppackage.cpp

```diff
@@ -36,7 +36,12 @@ using namespace Microsoft::WRL::Wrappers;
 
 extern "C" bool khrIcdOsVendorsEnumerateAppPackage()
 {
-    if (FAILED(Windows::Foundation::Initialize()))
+    HRESULT hrInit = Windows::Foundation::Initialize(RO_INIT_MULTITHREADED);
+    if (hrInit == RPC_E_CHANGED_MODE)
+    {
+        hrInit = Windows::Foundation::Initialize(RO_INIT_SINGLETHREADED);
```

Comment:
Does this only happen in cases where the application itself is expected to be single threaded?  I don't really understand the implications of falling back, or in what cases it would happen, but if the assumption above is true, seems good.

Reply:
Yeah, if the app has already initialized COM or WinRT as singlethreaded on the thread where the loader is being initialized, then we have to use singlethreaded as well. In all other cases (already initialized as multi-threaded, or uninitialized), we'll use multithreaded.